drivers/serial/Kconfig.hpmicro

```diff
@@ -0,0 +1,13 @@
+# Copyright 2022 hpmicro
+# SPDX-License-Identifier: Apache-2.0
+DT_COMPAT_HPM_UART := hpmicro,hpm-uart
```

Comment:
Using new style, this should become:

```
# Copyright 2022 hpmicro
# SPDX-License-Identifier: Apache-2.0

config UART_HPM
	bool "HPMICRO serial driver"
	default y
	depends on DT_HAS_HPMICRO_HPM_UART_ENABLED
	select USE_HPMSDK_UART
	select SERIAL_HAS_DRIVER
	select SERIAL_SUPPORT_INTERRUPT
	help
	  This option enables the UART driver for HPM SoC family.
```

(similar change needed for GPIO & Pinctrl Kconfig's)